### PR TITLE
feat: implement the excution of UDF Server call

### PR DESCRIPTION
**Pull request by gitccl**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Part of #12177.

In this pr, we implemented the bind and excution of UDF Server calls of some basic data types, like string and number data types. Here is a demo:
```sh
pip install pyarrow
# start UDF server
python3 tests/udf-server/udf.py
```

```sql
mysql> create function str_cmp as (string, string) -> int32 address='http://localhost:8815';
Query OK, 0 rows affected (0.03 sec)

mysql> select str_cmp('abc', 'abc'), str_cmp('abc', 'abd'), str_cmp('abc', 'aaa');
+-----------------------+-----------------------+-----------------------+
| str_cmp('abc', 'abc') | str_cmp('abc', 'abd') | str_cmp('abc', 'aaa') |
+-----------------------+-----------------------+-----------------------+
|                     0 |                    -1 |                     1 |
+-----------------------+-----------------------+-----------------------+
1 row in set (0.36 sec)
```

Now, there is no tests for udf server. I will add the python udf server tests into the sqllogic ci tests in the subsequent pr.

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12417)

### CODE REVIEW

**Comment by sundy-li** [comment]
> I will add the python udf server tests into the sqllogic ci tests in the subsequent pr.

I think you can wrote this server in rust and bind it into python in `src/bendpy`.

**Comment by gitccl** [comment]
> > I will add the python udf server tests into the sqllogic ci tests in the subsequent pr.
> 
> I think you can write this server in rust and bind it into python in `src/bendpy`.

Sorry, I didn't understand it. Do you mean write a ArrowFlight Server in rust code?

**Comment by sundy-li** [comment]
> > > I will add the python udf server tests into the sqllogic ci tests in the subsequent pr.
> > 
> > 
> > I think you can write this server in rust and bind it into python in `src/bendpy`.
> 
> Sorry, I didn't understand it. Do you mean write an ArrowFlight Server in rust code?

Yes, you can write the server into rust code and export friendly API to users in python code.

File: tests/udf-server/udf.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import pyarrow as pa
+import pyarrow.flight
+
+
+def str_cmp(s1: str, s2: str):
```
Comment (sundy-li) [inline tests/udf-server/udf.py:5]: it looks simple enough.

**Review by sundy-li — APPROVED** [review]

File: tests/udf-server/udf.py (hunk 1 above)
Comment (gitccl) [reply tests/udf-server/udf.py:5]: Yes. We only support simple types now. More types will be supported later.

**Review by xudong963 — APPROVED** [review]

**Pull request merged**